src/redux/rockets/rocketsSlice.js

```diff
@@ -0,0 +1,62 @@
+import axios from 'axios';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+
+const initialState = {
+  rocketItems: [],
+  isLoading: true,
+};
+
+export const getRocketItems = createAsyncThunk(
+  'rockets/getRocketItems',
+  async (_, thunkAPI) => {
+    try {
+      const response = await axios.get('https://api.spacexdata.com/v3/rockets');
+      return response.data;
+    } catch (error) {
+      return thunkAPI.rejectWithValue('Error...');
+    }
+  },
+);
+
+const rocketsSlice = createSlice({
+  name: 'rockets',
+  initialState,
+  reducers: {
+    reserveRocket: (state, action) => {
+      const newRocketItems = state.rocketItems.map((item) => {
+        if (item.id !== action.payload) { return item; }
+        if (item.reserved === true) {
+          return { ...item, reserved: false };
+        }
+        return { ...item, reserved: true };
+      });
+      state.rocketItems = newRocketItems;
+    },
+  },
+  extraReducers: (builder) => {
+    builder
+      .addCase(getRocketItems.fulfilled, (state, action) => {
+        state.isLoading = false;
```

Comment:
- Kindly, you are storing the whole API response object which is not necessary, Please filter and store only used data to redux store to improve its performance 👌
![2023-04-15_04-47](https://user-images.githubusercontent.com/52010789/232177487-2d5b4084-d8d9-4c28-a386-511166af9fc2.png)